tests/func/test_ls_url.py

```diff
@@ -0,0 +1,49 @@
+import os
+
+import pytest
+
+from dvc.exceptions import URLMissingError
+from dvc.repo import Repo
+
+
+def match_files(entries, expected):
+    entries_content = {(d["path"], d["isdir"]) for d in entries}
+    expected_content = {(d["path"], d["isdir"]) for d in expected}
+    assert entries_content == expected_content
+
+
+@pytest.mark.parametrize("fname", ["foo", "foo.dvc", "dir/foo"])
+def test_file(tmp_dir, fname):
+    tmp_dir.gen({fname: "foo contents"})
+
+    (d,) = Repo.ls_url(fname)
+    assert d["isdir"] is False
+    assert os.path.normpath(d["path"]) == os.path.abspath(fname)
+
+
+def test_dir(tmp_dir):
+    tmp_dir.gen({"dir/foo": "foo contents", "dir/subdir/bar": "bar contents"})
+
+    result = Repo.ls_url("dir")
+    match_files(
+        result,
+        [{"path": "foo", "isdir": False}, {"path": "subdir", "isdir": True}],
+    )
+
+
+def test_recursive(tmp_dir):
+    tmp_dir.gen({"dir/foo": "foo contents", "dir/subdir/bar": "bar contents"})
+
+    result = Repo.ls_url("dir", recursive=True)
+    match_files(
+        result,
+        [
+            {"path": "foo", "isdir": False},
+            {"path": "subdir/bar", "isdir": False},
+        ],
```

Comment:
Let's move these to https://github.com/iterative/dvc/blob/main/dvc/testing/test_workspace.py and use `workspace` instead of `tmp_dir` directly. It should be easy enough to do right away.

Reply:
Hmm, it doesn't seem so easy: the tests are specific to the local FS because they use `tmp_dir` and `os.path`. Also, `ls-url` doesn't interact with the workspace at all, so `workspace` probably doesn't help. The `cloud` fixture seems appropriate, but requires some work...

Reply:
@rlamy Gotcha. Then feel free to keep it as is for now.